packages/hash/api/src/db/postgres/client.ts

```diff
@@ -133,6 +140,18 @@ export class PostgresClient implements DBClient {
     });
   }
 
+  /**
+   * Conn is interchangable for allowing usage in transactions as well as standard connections.
+   */
+  static async getJsonSchemaBySchemaId(conn: Connection, schemaId: string) {
```

Comment:
The reason for this being static is that the connection could be a transaction, in which the `this.conn` property would be considered a parent transaction in slonik when [used here](https://github.com/hashintel/hash/blob/bd6ce30204e1dfb30ba1627b364d05370773bff1/packages/hash/api/src/db/postgres/client.ts#L95), causing an exception to be thrown akin to that of inserting concurrently through transactions.

This function is used for the AJV resolver, which makes me think that this might not even be the proper place for it to be located. I could construct a closure where the `JsonSchemaCompiler` is constructed in this version of the code and omit this static method completely, or pull it out into a module level const